CMakeLists.txt

```diff
@@ -85,8 +85,14 @@ if (NOT WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
     if (APPLE)
-        include_directories(BEFORE /usr/local/opt/openssl/include)
-        link_directories(/usr/local/opt/openssl/lib)
+        include_directories(BEFORE
+                /usr/local/opt/openssl/include
+                /opt/homebrew/opt/openssl@1.1/include
+                )
+        link_directories(
+                /usr/local/opt/openssl/lib
+                /opt/homebrew/opt/openssl@1.1/lib
+                )
```

Comment:
Can you please make a separate PR for this? There are two different fixes here. And also I recommend
```
if (APPLE)
    execute_process(COMMAND brew --prefix OUTPUT_VARIABLE BREW_PREFIX)
    message(STATUS "brew prefix ${BREW_PREFIX}")
    include_directories(BEFORE
        ${BREW_PREFIX}/opt/openssl@1.1/include
        /usr/local/opt/openssl/include
...
```

Reply:
You are right, I will do it later.